### DISCUSSION TITLE
Additional Dialogue

### DISCUSSION THREAD

**Discussion by TikkuTokku** [zippity21/Stardew_Valley_Regression_Mod#40] [Ideas]
I'm Tikku,

I got this mod a few days ago and got really excited about it. I didn't love the emphasis on humiliation, so I figured out how to change those lines. When I noticed there were very few villager reactions, I learned how to edit those too. Then I got really into it and learned how to edit dialogue options for just talking to villagers, including questions with branching options, and have already started working on ABDL specific lines. 

I would love to work on this with you if you're interested.

**Comment by zippity21** [comment]
That sounds awesome!

Part of the reason I put it on GitHub is so anybody can contribute. So I'd love to work with you.

**Reply by TikkuTokku** [reply]
Fantastic! I like the idea of discussing it on here. I don't want to step on your toes on anything, so I'll let you know what my ideas are and I'd love to hear how they work with your vision.

1)  I've changed a lot of the humiliation lines to be more supportive or little. For example, I changed, ""You feel humiliated knowing that you're still in ...", is now "You feel like a little <boy/girl> knowing that you're still in ..."
    I don't know if you preferred the humiliation text of it it was just what was already in the original mod. I can share those changes if you want to look them over. 

2)   I've been working on some character reactions to your wetting/messing. Mostly I've just learned the basic key formats for adding portraits or line breaks. So far I've only gotten one villagers reactions down before I got distracted with the next task I'd like to do.

3)   The thing I got really excited about learning (Which is also the reason I'm reaching out to you) is moding the character dialogue when you speak to a villager. I noticed Abigail had some reactions about her being messy, and it didn't make a lot of sense seeing that it came after you had just soiled yourself. Now I have it worked out where she can give that line if you have enough hearts with her. I've also figured out how to have a villager ask a question which can lead to some really cute ABDL scenes where you get to add context to how the villager is reacting to you. I'm very excited about that and am working through a few villagers right now.

4)   My big question for you is if you have any kind of vision for why Stardew Valley is very ABDL. Jodi's letter implies that it's something about the vally, which could be fun to explore. My friend wanted to change it to Jodi having learned about your little issue from your grandpa, which is cute, but it sets a very different scene. 
     An idea I'm exploring right now is having Vincent being a proud diaper boy and Jas being excited about pull ups. I'd also like Jodi to start giving very strong mommy vibes toward you as you befriend her. I also happened to make another friend of mine uncomfortable about having Vincent with ABDL lines (very understandable) So I'd also like to make all of that a supplement mod for those that are interested. 
     What are your thoughts on themes for the ABDL valley?

**Reply by zippity21** [reply]
I'll admit, my vision is limited. Since I was not the original creator, my initial goal was just to port it to 1.5.5 to help future proof it (and to get it playable again in general). Anything beyond that has been icing on the cake.

Though I am having fun with this, and its given me an excuse to learn C# (I know a few other languages, so thankfully I'm not starting from scratch, but still, there is always something new to learn). So I figure I'll maintain it and add to it for a while. But since I do lack a particular vision, I am looking towards the community to drive the creative aspects of this mod.

To that end, I'll try to actually address you list:

1. The humiliation stuff was definitely from the original. I think the pop-up text is supposed to be more of an internal dialogue, so I think feeling embarrassed it kind of natural, but I am personally more attracted to the idea of villagers being supportive. Though I do believe this should vary based on an individual's character personality and heart rating.

2. Linked to point 1. I Think the system for villager interaction/reactions could do with some reworking. I know that there are already existing bugs in relation to this (Issues #24 and #7 ).

3. I think that bug with Abigail was a carry-over from the original. I know someone on FoxTaleTimes had reported it to the previous maintainer. Though it is worth mentioning that the mod does have specialized dialogue for Abigail and Haley.

4. As addressed up-front, I am lacking in vision. You, of course are free to work on it here or fork to your own specialized version. My only concern becomes community desire. Unfortunately most users will not interact/provide feedback in a meaningful way (just how it goes), so its hard to know if changes are well-received. Though, it is perhaps better to not worry about it. If I clean up what's here to make sure the primary mechanics are working well, then do a release, then anybody who doesn't like the new tone can fall back on something. Thankfully, GitHub allows for multiple releases, so if there is something that really does need to be fixed, we can probably just release it with an appended title, like "_humiliation". Just my thoughts.

Having said all that, it sounds like you're rather proficient with the coding aspect, which is great! Are you also comfortable with Git. If not, the basics are quite easy. And if you are, then great again, you can get right to contributing and I'll look forward to your first Pull Request!

**Reply by TikkuTokku** [reply]
Thanks for the thorough reply! I'll take some creative liberties with the general tone and then be more than happy for notes when I've got a chunk done. I think I'm going to lean toward specific villagers hitting one particular button really hard. Abigale being already into diapers, Penny treating you like a child trying to potty train you, Haley teasing you, stuff like that. 

I actually have no experience with coding, but I learn fast and there are phenomenal tutorials out there for modding Stardew Valley. Right now I only have the skills to change dialogue using Content Patcher which turns out to be very easy. 

You mentioned a Pull Request. My guess is that would be me posting the code here to have others see if it's all good? How do I do that?

**Reply by zippity21** [reply]
Being new to modding, I was was actually completely unaware of Content Patcher. I looked it up and it seems pretty powerful. I'd actually be interested in migrating all the custom sprites and text to it since it seems so much more organized.

You have the general idea of a Pull Request correct, but it is somewhat more formal. You have to download a Git client, clone this repository, make a branch, apply your changes, commit it locally, push the branch and initiate a pull request here, describing your changes.

From there I review the changes, communicate with you to address any concerns and then it gets pulled into the main branch of code (where releases come from).

**Reply by zippity21** [reply]
If you need any help with doing that, I'd be more than happy to assist.

**Reply by TikkuTokku** [reply]
I'm hoping to get Jodi, Sam, and Vincent worked out before I send that over. With you're description I should be able to find a tutorial, and if I'm still having trouble I'll let you know!

**Reply by TikkuTokku** [reply]
First pull request!

**Reply by zippity21** [reply]
I've written a comment regarding the PR. I think we need to resolve it before moving forward. Content looks great though!
